packages/contracts-bedrock/contracts/test/CrossDomainOwnable3.t.sol

```diff
@@ -0,0 +1,151 @@
+        vm.prank(alice);
+        setter = new XDomainSetter3();
+
+        assertEq(setter.owner(), alice);
+    }
+
+    function test_localOnlyOwner_notOwner_reverts() public {
+        vm.prank(bob);
+        vm.expectRevert("CrossDomainOwnable3: caller is not the owner");
+        setter.set(1);
+    }
+
+    function test_crossDomainOnlyOwner_notOwner_reverts() public {
+        vm.expectEmit(true, true, true, true);
+        emit OwnershipTransferred(alice, alice, false);
+
+        vm.prank(setter.owner());
+        setter.transferOwnership({ _owner: alice, _isLocal: false });
+
+        // set the xDomainMsgSender storage slot
+        bytes32 key = bytes32(uint256(204));
+        bytes32 value = Bytes32AddressLib.fillLast12Bytes(bob);
+        vm.store(address(L2Messenger), key, value);
+
+        vm.prank(address(L2Messenger));
+        vm.expectRevert("CrossDomainOwnable3: caller is not the owner");
+        setter.set(1);
+    }
+
+    function test_crossDomainOnlyOwner_notOwner2_reverts() public {
+        vm.expectEmit(true, true, true, true);
+        emit OwnershipTransferred(alice, alice, false);
+
+        vm.prank(setter.owner());
+        setter.transferOwnership({ _owner: alice, _isLocal: false });
+
+        uint240 nonce = 0;
+        address sender = bob;
+        address target = address(setter);
+        uint256 value = 0;
+        uint256 minGasLimit = 0;
+        bytes memory message = abi.encodeWithSelector(XDomainSetter3.set.selector, 1);
+
+        bytes32 hash = Hashing.hashCrossDomainMessage(
+            Encoding.encodeVersionedNonce(nonce, 1),
+            sender,
+            target,
+            value,
+            minGasLimit,
+            message
+        );
+
+        // It should be a failed message. The revert is caught,
+        // so we cannot expectRevert here.
+        vm.expectEmit(true, true, true, true, address(L2Messenger));
+        emit FailedRelayedMessage(hash);
+
+        vm.prank(AddressAliasHelper.applyL1ToL2Alias(address(L1Messenger)));
+        L2Messenger.relayMessage(
+            Encoding.encodeVersionedNonce(nonce, 1),
+            sender,
+            target,
+            value,
+            minGasLimit,
+            message
+        );
+
+        assertEq(setter.value(), 0);
+    }
+
+    function test_crossDomainOnlyOwner_notMessenger_reverts() public {
+        vm.expectEmit(true, true, true, true);
+        emit OwnershipTransferred(alice, alice, false);
+
+        vm.prank(setter.owner());
+        setter.transferOwnership({ _owner: alice, _isLocal: false });
+
+        vm.prank(bob);
+        vm.expectRevert("CrossDomainOwnable3: caller is not the messenger");
+        setter.set(1);
+    }
+
+    function test_localOnlyOwner_succeeds() public {
+        vm.prank(setter.owner());
+        setter.set(1);
+        assertEq(setter.value(), 1);
+    }
+
+    function test_localTransferOwnership_succeeds() public {
+        vm.expectEmit(true, true, true, true);
```

Comment:
One thing that we are not catching here is the `OwnershipTransferred(address,address)` event in the base contract that emits when `super.TransferOwnership` is called. We probably want to have that in test coverage to prevent accidentally removing that and breaking things in the future

Reply:
addressed in the next commit!